src/tests/sqlsmith/src/runner.rs

```diff
@@ -13,15 +13,111 @@
+/// The goal is to generate NON-FAILING queries.
+/// If we encounter an expected error, just skip.
+/// If we panic or encounter an unexpected error, query generation
+/// should still fail.
+/// Returns ddl and queries.
+pub async fn generate(client: &tokio_postgres::Client, testdata: &str, count: usize, outdir: &str) {
+    let mut rng = rand::rngs::SmallRng::from_entropy();
+    let (tables, mviews, setup_sql) = create_tables(&mut rng, testdata, client).await;
+
+    test_sqlsmith(client, &mut rng, tables.clone(), &setup_sql).await;
+    tracing::info!("Passed sqlsmith tests");
+
+    let mut queries = String::with_capacity(10000);
+    set_distributed_query_mode(client).await;
+    let mut generated_queries = 0;
+    for _ in 0..count {
+        // ENABLE: https://github.com/risingwavelabs/risingwave/issues/7928
+        // test_session_variable(client, rng).await;
+        let sql = sql_gen(&mut rng, tables.clone());
+        tracing::info!("Executing: {}", sql);
+        let response = client.query(sql.as_str(), &[]).await;
+        let skipped = validate_response(&setup_sql, &format!("{};", sql), response);
+        if skipped == 0 {
+            generated_queries += 1;
+            queries.push_str(&format!("{};\n", &sql));
+        }
+    }
+    tracing::info!("Generated {} batch queries", generated_queries);
+
+    let mut generated_queries = 0;
+    for _ in 0..count {
+        // ENABLE: https://github.com/risingwavelabs/risingwave/issues/7928
+        // test_session_variable(client, rng).await;
+        let (sql, table) = mview_sql_gen(&mut rng, tables.clone(), "stream_query");
+        tracing::info!("Executing: {}", sql);
+        let response = client.query(&sql, &[]).await;
+        let skipped = validate_response(&setup_sql, &format!("{};", sql), response);
+        drop_mview_table(&table, client).await;
+        if skipped == 0 {
+            generated_queries += 1;
+            queries.push_str(&format!("{};\n", &sql));
+            queries.push_str(&format!("{};\n", format_drop_mview(&table)));
+        }
+    }
+    tracing::info!("Generated {} stream queries", generated_queries);
+
+    drop_tables(&mviews, testdata, client).await;
+    write_to_file(outdir, "ddl.sql", &setup_sql);
+    write_to_file(outdir, "queries.sql", &queries);
+}
+
+fn write_to_file(outdir: &str, name: &str, sql: &str) {
+    let resolved = format!("{}/{}", outdir, name);
+    let path = Path::new(&resolved);
+    let mut file = match File::create(path) {
+        Err(e) => panic!("couldn't create {}: {}", path.display(), e),
+        Ok(file) => file,
+    };
+    match file.write_all(sql.as_bytes()) {
+        Err(why) => panic!("couldn't write to {}: {}", path.display(), why),
+        Ok(_) => tracing::info!("successfully wrote to {}", path.display()),
+    }
+}
+
 /// e2e test runner for sqlsmith
 pub async fn run(client: &tokio_postgres::Client, testdata: &str, count: usize) {
     let mut rng = rand::rngs::SmallRng::from_entropy();
```

Comment:
Just came up with an idea: Can we make queries deterministic if we provide a fixed seed for this RNG?

The current `from_entropy` reads from the global RNG in the simulation test. If the execution path before this point changes (likely to happen when RW code developed), the initial state of this RNG changes as well. I guess that's why queries are not stable across PRs. 👀

Reply:
Okay let's try it first before this PR, that is simple change.

Reply:
https://github.com/risingwavelabs/risingwave/pull/8047